xonsh/vox.py

```diff
@@ -0,0 +1,197 @@
+            ('deactivate', 'exit'): self.deactivate,
+            ('list', 'ls'): self.list,
+            ('remove', 'rm', 'delete', 'del'): self.remove,
+            ('help', '-h', '--help'): self.help
+        }
+
+    def __call__(self, args, stdin=None):
+        """Call the right handler method for a given command."""
+
+        if not args:
+            self.help()
+            return None
+
+        command_name, params = args[0], args[1:]
+
+        try:
+            command = [
+                self.commands[aliases] for aliases in self.commands
+                if command_name in aliases
+            ][0]
+
+            command(*params)
+
+        except IndexError:
+            print('Command "%s" doesn\'t exist.\n' % command_name)
+            self.print_commands()
+
+    @staticmethod
+    def new(name):
+        """Create a virtual environment in $VIRTUALENV_HOME with python3's ``venv``.
+
+        :param name: virtual environment name
+        """
+
+        env_path = join(builtins.__xonsh_env__['VIRTUALENV_HOME'], name)
+
+        print('Creating environment...')
+
+        venv.create(env_path, with_pip=True)
+
+        print('Environment "%s" created. Activate it with "vox activate %s".\n' % (name, name))
+
+    @staticmethod
+    def activate(name):
+        """Activate a virtual environment.
+
+        :param name: virtual environment name
+        """
+
+        env_path = join(builtins.__xonsh_env__['VIRTUALENV_HOME'], name)
+
+        if not exists(env_path):
+            print('This environment doesn\'t exist. Create it with "vox new %s".\n' % name)
+            return None
+
+        if os.name == 'nt':
+            bin_dir = 'Scripts'
+
+        elif os.name == 'posix':
+            bin_dir = 'bin'
+
+        else:
+            print('This OS is not supported.')
+            return None
+
+        bin_path = join(env_path, bin_dir)
+
+        builtins.__xonsh_env__['PATH'].insert(0, bin_path)
+        builtins.__xonsh_env__['VIRTUAL_ENV'] = env_path
+
+        print('Activated "%s".\n' % name)
+
+    @staticmethod
+    def deactivate():
+        """Deactive the active virtual environment."""
+
+        if 'VIRTUAL_ENV' not in builtins.__xonsh_env__:
+            print('No environment currently active. Activate one with "vox activate".\n')
+            return None
+
+        env_path = builtins.__xonsh_env__['VIRTUAL_ENV']
+
+        env_name = basename(env_path)
+
+        if os.name == 'nt':
+            bin_dir = 'Scripts'
+
+        elif os.name == 'posix':
+            bin_dir = 'bin'
+
+        else:
+            print('This OS is not supported.')
+            return None
+
+        bin_path = join(env_path, bin_dir)
+
+        while bin_path in builtins.__xonsh_env__['PATH']:
+            builtins.__xonsh_env__['PATH'].remove(bin_path)
+
+        builtins.__xonsh_env__.pop('VIRTUAL_ENV')
+
+        print('Deactivated "%s".\n' % env_name)
+
+    @staticmethod
+    def list():
```

Comment:
Fixed in https://github.com/moigagoo/xonsh/commit/89d7b754fd4a5a63a298ba4a6b7401b5acdb3f37